src/utils/daos_hdlr.c

```diff
@@ -2226,11 +2201,21 @@ fs_copy_hdlr(struct cmd_args_s *ap)
 	daos_cont_info_t	dst_cont_info = {0};
 	struct file_dfs		src_file_dfs = {0};
 	struct file_dfs		dst_file_dfs = {0};
-	struct dm_args		ca = {0};
+	struct dm_args		*ca = NULL;
 	bool			is_posix_copy = true;
-	struct fs_copy_stats	num = {0};
+	struct fs_copy_stats	*num = NULL;
+
+	D_ALLOC(ca, sizeof(struct dm_args));
+	if (ca == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+	ap->dm_args = ca;
 
-	set_dm_args_default(&ca);
+	D_ALLOC(num, sizeof(struct fs_copy_stats));
+	if (num == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
```

Comment:
Will this leak ca?

Reply:
Actually, it won't leak, even without the explicit cleanup I just pushed. The reason is that over in [utils.go:freeCmdArgs()](https://github.com/daos-stack/daos/pull/11237/files#diff-2a10e44da826c23ea5b3390a8c3db666d1bd672512fd0d53e6f2d3626f290638R193), we free the members regardless of whether or not the handler succeeded. That's why the `fs copy` fi test didn't fail.